src/zjs_event.c

```diff
@@ -290,38 +305,33 @@ static jerry_value_t get_max_listeners(const jerry_value_t function_obj,
                                        const jerry_value_t argv[],
                                        const jerry_length_t argc)
 {
-    event_t *ev;
-
-    jerry_value_t event_emitter = zjs_get_property(this, HIDDEN_PROP("event"));
-    if (!jerry_get_object_native_handle(event_emitter, (uintptr_t*)&ev)) {
-        jerry_release_value(event_emitter);
-        ERR_PRINT("native handle not found\n");
-        return jerry_create_number(DEFAULT_MAX_LISTENERS);
+    if (argc) {
+        return zjs_error("invalid arguments");
     }
+    jerry_value_t event_emitter = zjs_get_property(this, HIDDEN_PROP("event"));
+    uint32_t max_listeners = get_max_event_listeners(event_emitter);
     jerry_release_value(event_emitter);
-    return jerry_create_number(ev->max_listeners);
+
+    return jerry_create_number(max_listeners);
 }
 
 static jerry_value_t set_max_listeners(const jerry_value_t function_obj,
                                        const jerry_value_t this,
                                        const jerry_value_t argv[],
                                        const jerry_length_t argc)
 {
-    event_t *ev;
-
-    jerry_value_t event_emitter = zjs_get_property(this, HIDDEN_PROP("event"));
-    if (!jerry_get_object_native_handle(event_emitter, (uintptr_t*)&ev)) {
-        jerry_release_value(event_emitter);
-        ERR_PRINT("native handle not found\n");
-        return jerry_acquire_value(this);
+    if (!jerry_value_is_number(argv[0])) {
+        return zjs_error("max listeners count must be first parameter");
     }
-    jerry_release_value(event_emitter);
+    jerry_value_t event_emitter = zjs_get_property(this, HIDDEN_PROP("event"));
 
-    if (!jerry_value_is_number(argv[0])) {
-        ERR_PRINT("max listeners count must be first parameter\n");
-        return jerry_acquire_value(this);
+    double num = jerry_get_number_value(argv[0]);
+    if (num < 0) {
+        return zjs_error("max listener value must be a positive integer");
     }
-    ev->max_listeners = (uint32_t)jerry_get_number_value(argv[0]);
+    zjs_obj_add_number(event_emitter, num, "maxListeners");
```

Comment:
This can be changed by the event module though. Its also hidden behind the hidden "event" property so its not accessible from the JS script.